spec/serializers/api/v3/ecf/school_serializer_spec.rb

```diff
@@ -51,16 +51,49 @@ module ECF
             end
           end
 
-          context "when the school is already in the cohort" do
-            let!(:school_cohort) { create(:school_cohort, school:, cohort:) }
-            it "returns the school cohort induction_programme_choice" do
-              expect(subject.serializable_hash[:data][:attributes][:induction_programme_choice]).to eq(school_cohort.induction_programme_choice)
+          context "when 'programme_type_changes_2025' feature is disabled" do
+            before { FeatureFlag.deactivate(:programme_type_changes_2025) }
+
+            %w[core_induction_programme full_induction_programme].each do |induction_programme_choice|
+              context "when the school is already in the cohort with '#{induction_programme_choice}'" do
+                let!(:school_cohort) { create(:school_cohort, school:, cohort:, induction_programme_choice:) }
+
+                it "returns the school cohort induction_programme_choice" do
+                  expect(subject.serializable_hash[:data][:attributes][:induction_programme_choice]).to eq(induction_programme_choice)
+                end
+              end
+            end
+
+            context "when school is not yet in the cohort" do
+              it "returns the an unknown induction programme choice" do
+                expect(subject.serializable_hash[:data][:attributes][:induction_programme_choice]).to eq("not_yet_known")
+              end
             end
           end
 
-          context "when school is not yet in the cohort" do
-            it "returns the an unknown induction programme choice" do
-              expect(subject.serializable_hash[:data][:attributes][:induction_programme_choice]).to eq("not_yet_known")
+          context "when 'programme_type_changes_2025' feature is enabled" do
+            before { FeatureFlag.activate(:programme_type_changes_2025) }
+
+            context "when the school is already in the cohort with 'core_induction_programme'" do
```

Comment:
same here around the mappings please:
```
design_our_own
school_funded_fip
no_early_career_teachers
```
where `no_early_career_teachers` should also be returned the same